sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md

```diff
@@ -0,0 +1,647 @@
+| `(acc-*)` | ([Stateful](#statelessstateful-memory-mode) memory mode: `-fno-sycl-esimd-force-stateless-mem`) and (no cache-hints) and (`pred` is not passed) and (sizeof(`T`) * `N` == 16,32,64,128) | sizeof(`T`)*`N` == 16,32,64,128 | Any Intel GPU |
+| `(acc-*)` | ([Stateful](#statelessstateful-memory-mode) memory mode: `-fno-sycl-esimd-force-stateless-mem`) and ((cache-hints) or (`pred` is passed)) or (sizeof(`T`) * `N` != 16,32,64,128) | `N` must be from [Table2 below](#table1---valid-values-of-n-if-cache-hints-used-or-pred-parameter-is-passed) | DG2 or PVC |
+| `(lacc-1,2)`, `(slm-2)`  | `pred` is not passed | `N` is any positive number | Any Intel GPU |
+| `(lacc-3,4)`, `(slm-1)`  | `pred` is passed | `N` must be from [Table2 below] | DG2 or PVC |
+
+
+#### Table2 - Valid values of `N` if cache-hints used or `pred` parameter is passed:
+| sizeof(`T`) | Valid values of `N` | Special case - PVC only - the maximal `N`: requires bigger alignment: 8 or more |
+|---|--------------------------------|-----|
+| 1 | 4, 8, 12, 16, 32, 64, 128, 256 | 512 |
+| 2 | 2, 4, 6, 8, 16, 32, 64, 128 | 256 |
+| 4 | 1, 2, 3, 4, 8, 16, 32, 64 | 128 |
+| 8 | 1, 2, 3, 4, 8, 16, 32 | 64 |
+
+## gather(...) - load from memory locations addressed by a vector of offsets
+```C++
+namespace sycl::ext::intel::esimd {
+// gather from USM memory - general form (must specify T, N, VS parameters).
+template <typename T, int N, int VS, typename OffsetT, typename PropertyListT = empty_properties_t>
+/*usm-1*/ simd<T, N> gather(const T *p, simd<OffsetT, N / VS> byte_offsets, simd_mask<N / VS> mask,
+                            simd<T, N> pass_thru, PropertyListT props = {});
+/*usm-2*/ simd<T, N> gather(const T *p, simd<OffsetT, N / VS> byte_offsets, simd_mask<N / VS> mask,
+                            PropertyListT props = {});
+/*usm-3*/ simd<T, N> gather(const T *p, simd<OffsetT, N / VS> byte_offsets,
+                            PropertyListT props = {});
+
+// gather from USM - convenience/short form (VS = 1; T and N can also be omitted)
+template <typename T, int N, typename OffsetT, typename PropertyListT = empty_properties_t>
+/*usm-4*/ simd<T, N> gather(const T *p, simd<OffsetT, N> byte_offsets, simd_mask<N> mask, simd<T, N> pass_thru,
+                            PropertyListT props = {});
+/*usm-5*/ simd<T, N> gather(const T *p, simd<OffsetT, N> byte_offsets, simd_mask<N> mask,
+                            PropertyListT props = {});
+/*usm-6*/ simd<T, N> gather(const T *p, simd<OffsetT, N> byte_offsets,
+                            PropertyListT props = {});
+
+// gather from USM - general form accepting offsets as simd_view
+template <typename T, int N, int VS = 1, typename OffsetObjT,
+          typename OffsetRegionT, typename PropertyListT = empty_props_t>
+/*usm-7*/ simd <T, N> gather(const T *p, simd_view<OffsetObjT, OffsetRegionT> byte_offsets,
```

Comment:
Should these be OffsetSimdViewT?

Reply:
In general yes, but the doc only shows what we have in memory.hpp for (usm-ga-7)

Reply:
We should probably have a separate internal tracker to fix all those